src/components/TaskQueues.tsx

```diff
@@ -1,14 +1,85 @@
-import React, { useState } from 'react';
-import { taskQueuesData } from '../data/data';
+import React, { useEffect, useState, Dispatch, SetStateAction } from 'react';
 import { BsThreeDots } from 'react-icons/bs';
 import { FaChevronDown } from 'react-icons/fa';
-import AdminEditTask from './AdminEditTask';
+import AdminModifyTaskQueue from './AdminModifyTaskQueue';
 import Modal from './Modal';
 
+const workspaceSid = process.env.NEXT_PUBLIC_WORKSPACE_SID as string
+
+interface IProps{
+  sid: number,
+  taskQueueName: string
+  setShowModal: Dispatch<SetStateAction<boolean>>;
+}
+
 const TaskQueuesTable = () => {
   const [showSortOptions, setShowSortOptions] = useState(false);
   const [modalContent, setModalContent] = useState<React.ReactNode>(null);
   const [showModal, setShowModal] = useState(false);
+  const [taskQueuesData, setTaskQueuesData] = useState([]);
+  const [assignedUsers, setAssignedUsers] = useState([[""]]);
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  async function getAssignedUsers(taskQueues: [{friendlyName: string}]) {
+
+    const getWorkers = await fetch(`/api/workers?workspaceSid=${process.env.NEXT_PUBLIC_WORKSPACE_SID}`,{
+      method:'GET'
+    })
+    const workersResponse = await getWorkers.json()
+    const workersList = workersResponse.workers
+
+    var allAssignedUsers = []
+
+    for(let i=0; i < taskQueues.length; i++){
+
+      var curAssignedUsers = []
+    
+      for(let j=0; j < workersList.length; j++){
+        const attributes = JSON.parse(workersList[j].attributes)
+        if(attributes.hasOwnProperty("taskQueues")){
+          if(attributes.taskQueues.includes(taskQueues[i].friendlyName)){
+            curAssignedUsers.push(workersList[j].friendlyName)
+          }
+        }
+      }
+      if (curAssignedUsers.length === 0){
+        curAssignedUsers.push("No Workers Assigned")
```

Comment:
Could we refactor this logic so that if curAssignedUsers is empty, we show that message in the UI?

Reply:
This already works :)

Reply:
LMAO okok fine we'll keep it like this xD